internal/proxy/README.md

```diff
@@ -17,16 +17,42 @@ dependency to the proxy:
 1.  Ensure that the new dependency is covered under one of the `notice`,
     `permissive`, or `unencumbered` licences under the categories defined by
     [Google Open Source](https://opensource.google.com/docs/thirdparty/licenses/).
-2.  Add the dependency to the GCS bucket:
+2.  Gather the new set of dependencies and sync them to the GCS bucket.
 
 ```bash
-./internal/proxy/makeproxy.sh proxy
+# This approach was suggested by:
+# https://github.com/go-modules-by-example/index/tree/master/012_modvendor
+
+# Create a temporary directory where we'll create a module download cache.
+tgp=$(mktemp -d)
+
+# Run this command in the master branch.
+# It runs "go mod download" in every module that we have in our repo,
+# filling the cache with all of the module dependencies we need.
+GOPROXY=file://$GOPATH/pkg/mod/cache/download GOPATH=$tgp ./internal/proxy/makeproxy.sh
```

Comment:
Seems like this environment variable juggling could happen in the script. I would want this instruction to read:

```
./internal/proxy/makeproxy.sh "$tgp"
```

Reply:
That would be nice, but the fact that you have to run "go mod download" from multiple branches makes it difficult... (see the next comment).

Reply:
I don't see the relation here: what about multiple branches is incompatible with setting the environment variables within the script instead of outside it?

Reply:
Oh, I see what you're saying; done.